Source/Particles/NamedComponentParticleContainer.H

```diff
@@ -0,0 +1,120 @@
+#ifndef NamedComponentParticleContainer_H_
+#define NamedComponentParticleContainer_H_
+
+#include <AMReX.H>
+#include <AMReX_Vector.H>
+#include <AMReX_AmrParticles.H>
+#include <AMReX_Particles.H>
+#include <AMReX_ParIter.H>
+#include "Utils/TextMsg.H"
+
+struct PIdx
```

Comment:
This `struct` used to be in `WarpXParticleContainer`.

Reply:
Generally speaking, I would like to move this into its own file in WarpX.

That way, we could move `NamedComponentParticleContainer` into ABLASTR.